maven-plugin/src/main/java/org/openmrs/maven/plugins/model/SdkStatistics.java

```diff
@@ -0,0 +1,325 @@
+package org.openmrs.maven.plugins.model;
+
+import com.dropbox.core.DbxException;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.openmrs.maven.plugins.utility.SDKConstants;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Properties;
+import java.util.UUID;
+
+/**
+ *
+ */
+public class SdkStatistics {
+
+    private static final String SDK_STATS_FILE_NAME = "sdk-stats.properties";
+    private static final String DATE_FORMAT = "dd-M-yyyy";
+
+    private Properties statistics = new Properties();
+
+    public SdkStatistics createSdkStatsFile(boolean agree) throws MojoExecutionException {
+        loadStatsFromResource();
+        initData();
+        setStatsEnabled(agree);
+        save();
+        return new SdkStatistics(statistics);
+    }
+
+    public static SdkStatistics loadStatistics() throws IllegalStateException, MojoExecutionException {
+        return new SdkStatistics(new File(Server.getServersPath(), SDK_STATS_FILE_NAME));
+    }
+
+    public SdkStatistics(){}
+
+    private SdkStatistics(Properties statistics) throws MojoExecutionException {
+        this.statistics = statistics;
+    }
+
+    SdkStatistics(File file) throws IllegalStateException, MojoExecutionException {
+        if (file.exists()) {
+            loadStatsFromFile(file);
+        } else {
+            throw new IllegalStateException("File does not exist");
+        }
+        setSdkVersion(SDKConstants.getSDKInfo().getVersion());
+    }
+
+    /**
+     * Checks if there is more then 7 days since last sendReport. If yes, send statistics
+     * @throws MojoExecutionException
+     */
+    public void sendReport() throws MojoExecutionException {
+        if( checkIfOneWeekFromLastReport() && getStatsEnabled()){
+            setLastReported();
+            save();
+            try {
+                postToGoogleForm();
+            } catch (DbxException | IOException e) {
+                throw new IllegalStateException("Could not send statistics", e);
```

Comment:
I'd rather we fail silently if stats cannot be sent... Let's just display a short message to the user "Anonymous usage statistics could not be sent due to " +  e.getMessage() and continue execution.